python/GafferUI/CompoundEditor.py

```diff
@@ -52,52 +52,88 @@
 
 class CompoundEditor( GafferUI.Editor ) :
 
+	class _DetachedPanel( GafferUI.Window ) :
+
+		def __init__( self, scriptNode, *args, **kwargs ) :
+
+			GafferUI.Window.__init__( self, *args, **kwargs )
+			self.__titleManger = GafferUI.ScriptWindow._WindowTitleBehaviour( self, scriptNode )
+
+			self.__splitContainer = _SplitContainer()
+			self.__splitContainer.append( _TabbedContainer() )
+			self.setChild( self.__splitContainer )
+
+			self._gafferParent = None
+
+		## As were technically not in the Qt hierarchy, but do want to be logically
+		# owned by a CompoundEditor, we re-implement this to re-direct ancestor
+		# calls to which ever editor we were added to (CompoundEditor fills this
+		# in for us when we're added).
+		def parent( self ) :
+
+			return self._gafferParent() if self._gafferParent else None
+
+		def editors( self, type = GafferUI.Editor ) :
+
+			return self.__splitContainer.editors( type = type )
+
+		def addEditor( self, editor ) :
+
+			self.__splitContainer.addEditor( editor )
+
+		def replaceEditors( self, splitContainer ) :
+
+			self.__splitContainer = splitContainer
+			self.setChild( self.__splitContainer )
+
+		# A detached panel is considered empty if it has a single split with
+		# no editors.
+		def isEmpty( self ) :
+
+			if len(self.__splitContainer) == 1 :
+				c = self.__splitContainer[0]
+				if isinstance( c, _TabbedContainer ) and len(c) == 0 :
+					return True
+
+			return False
+
+		def _setGafferParent( self, gafferWidget ) :
```

Comment:
Because of the serialisation I don't think its straight forward to pass it on construction, so it gets set when the DetachedPanel is added to an Editor. Any thoughts on other ways to do that?

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/3167/commits/4361ca224e657be1db373d6d0fe91d18a252c235

Reply:
Note: We initially discussed not having a factory method on `CompoundEditor`, but it made all the other code that made panels a mess, so went for one in the end.